### PR TITLE
[storage-blob-preview] Container soft delete 

### PR DESCRIPTION
**Pull request by Juliehzl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
storage-blob-preview

**Comment by zhoxing-ms** [comment]
It's curious that the new commands are `storage container list` and `storage container restore`. Why the name of PR is `Container soft delete`?

**Review by jsntcy — APPROVED** [review]
:shipit:

File: src/storage-blob-preview/azext_storage_blob_preview/_params.py [hunk 1]
```diff
@@ -465,3 +468,25 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
+        c.argument('include_deleted', arg_type=get_three_state_flag(), min_api='2020-02-10',
+                   help='Specify that deleted containers to be returned in the response. This is for container restore '
+                   'enabled account. The default value is `False`')
+
+    with self.argument_context('storage container restore') as c:
+        c.argument('deleted_container_name', options_list=['--name', '-n'],
+                   help='Specify the name of the deleted container to restore.')
+        c.argument('deleted_container_version', options_list=['--deleted-version'],
+                   help='Specify the version of the deleted container to restore.')
+        c.argument('new_name', help='The new name for the deleted container to be restored to.')
```
Comment (zhoxing-ms) [inline src/storage-blob-preview/azext_storage_blob_preview/_params.py:491]: Could we consider adding a test for this parameter?
By the way, may I ask `deleted_container_name` and `new_name` can have the same value?

**Pull request merged**